### DISCUSSION TITLE
Queue Manager, Notification and Daemon not running

### DISCUSSION THREAD

**Discussion by marcus-at-localhost** [atrocore/atrocore#185] [Q&A] [answered]
I setup the cronjob and it seems to be running (/data/cache/application/cronLastRunTime.php is changing every minute and the apidocs are generated every 2h)

What's not working is the Queue and the Notification for Export.

I upload an image, everything looks fine incl. Thumbnails are generated - yet I got a "Pending for Generating Thumnail  xy.jpg" in the Queue Manager. Same when I start an Export.

I started the Cli script manually with `/usr/bin/php7.3-cli htdocs/atropim/index.php qm Export --run` (not sure if `Export` is correct here, probably not :D) and I was able to empty the queue row by row.

What do I have to check if the cronjob and all the other jobs are setup correctly (Some DB Table/Entry, files)? Is there a way to debug the cronjob (write a logfile)?

And since it's recommended to update from the Module Manger - I get the warning that the Daemon is not running.

Thanks for your assistance!

**Comment by marcus-at-localhost** [comment, marked as answer]
Apparently in my case the wrong php cli version is assumed here https://github.com/atrocore/atrocore/blob/master/app/Treo/Console/Cron.php#L78

or better here https://github.com/atrocore/atrocore/blob/master/app/Espo/Core/Utils/System.php#L117 

It's detecting `/usr/bin/php` but in my case the correct php cli version is `/usr/bin/php7.4-cli`

So everything afterwards seems to fail silently :/

I corrected the path and daemon, notifications etc work. 

Can there be a config setting to set the correct php cli version in case it is different?

**Reply by rratsun** [reply]
We added this issue to our backlog and will do it soon.

**Reply by marcus-at-localhost** [reply]
For the record, this was solved here: https://github.com/atrocore/atrocore/issues/109#issuecomment-844455166

> For this you should set phpBinPath parameter in data/config.php.
> 
> Example:
> 
> ```php
> 'phpBinPath' => 'php',
> ```

**Comment by Yurpal** [comment]
Hello, 

 I am having a similar issue 
 
We are using SiteGround so have to set a Cron Job in the UI 

/usr/local/bin/php /home/customer/www/vendors.yurpal.com/public_html/index.php 

We are still unable to have the cron run properly. Do you know why this is?

**Reply by atrolex** [reply]
<img width="475" alt="cron" src="https://user-images.githubusercontent.com/33658481/117531918-35ce0d80-afe5-11eb-8bad-1672ef7faf9a.png">
"cron" should be stated in your cron job, it is important.

**Reply by Yurpal** [reply]
Thank you! Also - the error log I am seeing 

PHP Fatal error: The allowed memory size of 2097152 bytes exhausted (tried to allocate 8192 bytes) in /home/customer/www/vendors.yurpal.com/public_html/vendor/atrocore/core/app/Espo/Core/Utils/Json.php on line 75 

The memory limit has been increased twice. What would be blocking the allowed memory size?
